### PR TITLE
better error handling on backfill creation

### PR DESCRIPTION
**Pull request by eryshev**
- now we can backfill a hourly job for one period only with its daily dependencies being properly backfilled
- now we have one precise error by each check on backfill creation
- errors are displayed in a proper UI widget
- I extract some UI components that I thought I can reuse for backfill's UI but at the end I didn't use them. However I consider this change as a good practice for React as a way to have smaller components.

### CODE REVIEW

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala [hunk 1]
```diff
@@ -289,47 +289,95 @@ case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] wit
...
                                       description: String,
                                       jobs: Set[TimeSeriesJob],
                                       start: Instant,
                                       end: Instant,
                                       priority: Int,
-                                      xa: XA)(implicit user: Auth.User): IO[Boolean] = {
-    val (isValid, newBackfill) = atomic { implicit txn =>
-      val id = UUID.randomUUID().toString
-      val newBackfill = Backfill(id, start, end, jobs, priority, name, description, "RUNNING", user.userId)
+                                      xa: XA)(implicit user: Auth.User): IO[Either[String, Boolean]] = {
```
Comment (haveo) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala]: Why not `IO[Either[String, Unit]]`?

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala [hunk 2]
```diff
@@ -289,47 +289,95 @@ case class TimeSeriesScheduler(logger: Logger) extends Scheduler[TimeSeries] wit
...
-      }
-      (isValid, newBackfill)
-    }
-    if (isValid)
-      Database.createBackfill(newBackfill).transact(xa).map(_ => true)
+    if (isBackfillValid)
+      Database.createBackfill(newBackfill).transact(xa).map(_ => Right(true))
     else
-      IO.pure(false)
+      IO.pure(Left(validationByJob.filter(_.isLeft).map(_.left.get).mkString("\n")))
```
Comment (haveo) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesScheduler.scala]: `validationByJob.collect { Left(x) => x }.mkString("\n")`

**Comment by eryshev** [comment]
You are right. I corrected it and added a new functionality.

File: timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesApp.scala [hunk 3]
```diff
@@ -528,10 +536,17 @@ private[timeseries] trait TimeSeriesApp { self: TimeSeriesScheduler =>
...
                               backfill.endDate,
                               backfill.priority,
-                              xa).map({
-                    case true => Ok("ok".asJson)
-                    case _    => BadRequest("invalid backfill")
-                  })
+                              xa).map {
+                    case Right(_)     => Ok("ok".asJson)
+                    case Left(errors) => BadRequest(errors)
+                    case _ =>
```
Comment (haveo) [inline timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesApp.scala]: I don't think you need this anymore

Comment (eryshev) [reply timeseries/src/main/scala/com/criteo/cuttle/timeseries/TimeSeriesApp.scala]: Yep

**Comment by haveo** [comment]
LGTM (backend only)

File: core/src/main/javascript/app/pages/BackfillCreate.js [hunk 4]
```diff
@@ -128,7 +132,7 @@ const DateWithMask = ({
   </span>
 );
 
-class BackfillCreate extends React.Component<any, Props, void> {
+class BackfillCreate extends React.Component<any, Props, any> {
```
[the remark this answers is not in the document]
Comment (eryshev) [reply core/src/main/javascript/app/pages/BackfillCreate.js]: Yes, void :)

File: core/src/main/javascript/app/pages/Execution.js [hunk 5]
```diff
@@ -196,64 +214,21 @@ class Execution extends React.Component {
...
-              >
-                Cancel
-              </span>
-            ]}
-          />
-        );
-      }
-      return null;
-    };
+    const status =
```
[the remark this answers is not in the document]
Comment (eryshev) [reply core/src/main/javascript/app/pages/Execution.js]: Agree

File: core/src/main/javascript/app/pages/Execution.js [hunk 6]
```diff
@@ -21,28 +17,27 @@ import Status from "../components/Status";
...
+import TextWithDashedLine from "../components/TextWithDashedLine";
 
 type Line = {
   timestamp: string,
   level: "DEBUG" | "INFO" | "ERROR",
   message: string
 };
 
-type Props = {
+type ExecutionProps = {
```
[the remark this answers is not in the document]
Comment (eryshev) [reply core/src/main/javascript/app/pages/Execution.js]: Agree

**Comment by eryshev** [comment]
Updated this PR after @loicknuchel 's review.

**Pull request merged**